c_src/adbc_nif.cpp

```diff
@@ -451,6 +540,84 @@ int arrow_array_to_nif_term(ErlNifEnv *env, struct ArrowSchema * schema, struct
             children_term = enif_make_list_from_array(env, children.data(), (unsigned)schema->n_children); 
         } else if (format_len > 4 && (strncmp("+ud:", format, 4) == 0 || strncmp("+us:", format, 4) == 0)) {
             children_term = get_arrow_array_union_children(env, schema, values, level);
+        // date
+        } else if (strncmp("td", format, 2) == 0) {
+            char unit = format[2];
+
+            using value_type = uint64_t;
+            current_term = values_from_buffer(
+                env,
+                values->length,
+                bitmap_buffer,
+                (const value_type *)values->buffers[1],
+                [unit](ErlNifEnv *env, uint64_t val) -> ERL_NIF_TERM {
+                    switch (unit) {
+                        case 'D': // days
+                            return unix_second_offset_to_date(env, val * 24 * 60 * 60);
+                        case 'm': // milliseconds
+                            return unix_second_offset_to_date(env, val / 1000);
+                    }
+                }
```

Comment:
This generates a warning:

```plain
adbc/c_src/adbc_nif.cpp:560:17: warning: non-void lambda does not return a value in all control paths [-Wreturn-type]
                }
                ^
1 warning generated.
```

Should there be some sort of explicit handling for invalid formats? Same for the other `switch` statements for time/timestamp.

Reply:
Just set `format_processed = false` and we are good to go.